gslab_scons/tests/test_build_lyx.py

```diff
@@ -1,47 +1,107 @@
 #! /usr/bin/env python
-
 import unittest
 import sys
 import os
 import shutil
+import mock
+import re
+# Import gslab_scons testing helper modules
+import _test_helpers as helpers
+import _side_effects as fx
 
 # Ensure that Python can find and load the GSLab libraries
 os.chdir(os.path.dirname(os.path.realpath(__file__)))
 sys.path.append('../..')
 
-from gslab_scons import build_lyx
+import gslab_scons.builders.build_lyx as gs
 from gslab_scons._exception_classes import BadExtensionError
-from gslab_make.tests import nostderrout
+
+# Define path to the builder for use in patching
+path = 'gslab_scons.builders.build_lyx'
 
 
-class test_build_lyx(unittest.TestCase):
+class TestBuildLyX(unittest.TestCase):
 
     def setUp(self):
         if not os.path.exists('./build/'):
             os.mkdir('./build/')
 
-    def test_default(self):
-        env = ''
-        build_lyx('./build/lyx.pdf', './input/lyx_test_file.lyx', env)
-        logfile_data = open('./build/sconscript.log', 'rU').read()
-        self.assertIn('Log created:', logfile_data)
-        if os.path.isfile('./build/sconscript.log'):
-            os.remove('./build/sconscript.log')
-    
-    def test_bad_extension(self):
-        '''Test that build_lyx() recognises an inappropriate file extension'''
-        env = ''
-        with self.assertRaises(BadExtensionError), nostderrout():
-            build_lyx('./build/lyx.pdf', 
-                      ['bad', './input/lyx_test_file.lyx'], env)
+    @mock.patch('%s.os.system' % path)
+    def test_default(self, mock_system):
+        '''
+        Test that build_lyx() behaves correctly when provided with
+        standard inputs. 
+        '''
+        mock_system.side_effect = fx.lyx_side_effect
+        target = './build/lyx.pdf'
+
+        helpers.standard_test(self, gs.build_lyx, 'lyx', 
+                              system_mock = mock_system, target = target)
+        self.assertTrue(os.path.isfile(target))
+
+    @mock.patch('%s.os.system' % path)
+    def test_list_arguments(self, mock_system):
+        '''
+        Check that build_lyx() works when its source and target 
+        arguments are lists
+        '''
+        mock_system.side_effect = fx.lyx_side_effect
+        target = './build/lyx.pdf'
+
+        helpers.standard_test(self, gs.build_lyx, 'lyx', 
+                              system_mock = mock_system, target = target)
+        self.assertTrue(os.path.isfile(target))
+
+    test_bad_extension = \
```

Comment:
No good reason... I replaced all lambdas like this with normal function definitions.